core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -561,5 +568,17 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
       case _: TopicAuthorizationException => consumeRecords(consumer, numRecords, startingOffset, topic)
     }
   }
+
+  // Consume records, ignoring TopicAuthorization exception from previously sent request.
+  private def consumeRecordsIgnoreAuthorizationException(consumer: Consumer[Array[Byte], Array[Byte]],
+    numRecords: Int = 1,
+    startingOffset: Int = 0,
+    topic: String = topic): Unit = {
+    try {
+      consumeRecords(consumer, numRecords, startingOffset, topic)
+    } catch {
+      case _: TopicAuthorizationException => consumeRecordsIgnoreAuthorizationException(consumer, numRecords, startingOffset, topic)
```

Comment:
Is this recursive call necessary? TestUtils.waitUntilTrue already contains a loop, so perhaps we can streamline the check as follows:

```
      TestUtils.waitUntilTrue(() => {
        try {
          consumeRecords(consumer)
          true
        } catch {
          case _: TopicAuthorizationException => false
        }
      }, "Consumer didn't manage to consume the records within timeout.")
```